registry/manifest.go

```diff
@@ -149,32 +162,46 @@ func (registry *Registry) ImageIndex(repository, reference string) (*manifestlis
 }
 
 func (registry *Registry) ManifestOCI(repository, reference string) (*ocischema.DeserializedManifest, error) {
+	deserialized, _, err := registry.ManifestOCIWithDigest(repository, reference)
+	return deserialized, err
+}
+
+// ManifestOCIWithDigest extends ManifestOCI to return the digest found in the Docker-Content-Digest header.
+// If the header does not exist or is invalid an empty digest is returned (an error is not returned
+// so that existing clients are not affected by new, potentially unrelated, errors).
+func (registry *Registry) ManifestOCIWithDigest(repository, reference string) (*ocischema.DeserializedManifest, digest.Digest, error) {
```

Comment:
I'm wondering if we should switch the order, similar to `ManifestDigest`: `(digest.Digest, *ocischema.DeserializedManifest, error)`

Reply:
I believe it currently follows the convention for the primary concern to be the first return.

For `func ManifestOCIWithDigest() (*ocischema.DeserializedManifest, digest.Digest, error)`, the primary concern is the manifest, secondary the digest.

For `func ManifestDigest() (digest.Digest, string, error)` primary concern is the digest, secondary is the content type.

If you feel strongly about this let me know and I'll change it, seems to me to be appropriate as is.